add initial logging setup for testing against GrayLog

adamkorynta opened the pull request.

adamkorynta committed:
add initial logging setup for testing against GrayLog

adamkorynta committed:
add in docker files used for testing loki/graylog

adamkorynta committed:
updates to startServ to get working from $CWMS_HOME override

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/cwmsClasspath:
```diff
@@ -0,0 +1,61 @@
+This is useful for interactively checking the contents of the classes and jars
+directories by executing "cwmsClasspath /check"
+
+Mike Perryman
```
Do you want to keep this header comments as is including referencing Mike P

adamkorynta replied:
Updated

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/cwmsClasspath:
```diff
@@ -0,0 +1,61 @@
+'''
+
+import sys, os
+from fnmatch import fnmatch
```
what uses this import?

adamkorynta replied:
nothing, removed

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/cwmsClasspath:
```diff
@@ -0,0 +1,61 @@
+	print( "" )
+	sys.exit()
+
+import subprocess
```
why import here instead of consolidating them above?

adamkorynta replied:
Moved up to the top

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/cwmsClasspath:
```diff
@@ -0,0 +1,61 @@
+
+classpath = subprocess.check_output([os.path.join(cwmsExe, 'bin/cwmsClasspath')])
+if classpath:
+	print(classpath + ":" + cwmsHome + "/bin/lib/logstash-gelf-1.15.1.jar")
```
use a variable for logstash to make updates easier?

adamkorynta replied:
added

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/logWatcher.py:
```diff
@@ -0,0 +1,50 @@
+import os
```
assuming this script is going to be executed, then first line should call out the interpreter for this file:
#!/bin/env python

adamkorynta replied:
added

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/loggingSetup.py:
```diff
@@ -0,0 +1,44 @@
+#  Copyright (c) 2024
```
Why did you separate this from logWatcher? What is the re-use?

rma-psmorris replied:
@adamkorynta

adamkorynta replied:
loggingSetup.py can be used by any python script, such as instapost

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/logWatcher.py:
```diff
@@ -0,0 +1,50 @@
+
+    try:
+        while True:
+            time.sleep(1)
```
what unit is 1?

adamkorynta replied:
Seconds, made into a named variable

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/loggingSetup.py:
```diff
@@ -0,0 +1,44 @@
+        (logging.getLogger("logging_setup")
+         .warning("Could not load python logging config. CWMS_HOME not defined."))
+    else:
+        path = cwms_home + '/config/system_config/logging_config.json'
```
do you want to prefix this as python logging config?

adamkorynta replied:
Good idea, updated

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/sendLogMessage:
```diff
@@ -0,0 +1,81 @@
+
+LOGGING_HTTP_ENDPOINT=$(getProp cwms.enterprise.logging.http.url notFound cwms.properties $CWMS_HOME/config/properties)
+LOG_HOST=$(hostname)
+LOG_FACILITY='CWMS'
```
why is this set to CWMS?

adamkorynta replied:
well, the documentation no longer says since it's deprecated. Removed

rma-psmorris commented on cwms-servers/CWMS_HOME/bin/sendLogMessage:
```diff
@@ -0,0 +1,81 @@
+                          \"_process_name\": \"$PROCESS_NAME\"
+                      }"
+  #echo $PAYLOAD
+  curl -X POST -H 'Content-Type: application/json' -s -d "$PAYLOAD" $LOGGING_HTTP_ENDPOINT
```
this is https?

adamkorynta replied:
depends on how the cwms.properties is configured

rma-psmorris commented on cwms-servers/CWMS_HOME/config/system_config/cwms_system_logging.properties:
```diff
@@ -0,0 +1,61 @@
+biz.paluch.logging.gelf.jul.GelfLogHandler.host = http:<host>
+biz.paluch.logging.gelf.jul.GelfLogHandler.port = <port>
+biz.paluch.logging.gelf.jul.GelfLogHandler.facility = "NWP CWMS v3.3"
+biz.paluch.logging.gelf.jul.GelfLogHandler.level = FINE
```
Only FINE and not ALL?

adamkorynta replied:
Set to ALL to match the console handler

rma-psmorris commented on cwms-servers/CWMS_HOME/config/system_config/cwms_system_logging.properties:
```diff
@@ -0,0 +1,61 @@
+biz.paluch.logging.gelf.jul.GelfLogHandler.port = <port>
+biz.paluch.logging.gelf.jul.GelfLogHandler.facility = "NWP CWMS v3.3"
+biz.paluch.logging.gelf.jul.GelfLogHandler.level = FINE
+biz.paluch.logging.gelf.jul.GelfLogHandler.maximumMessageSize = 5000
```
Does this include stacks?

adamkorynta replied:
yes, but if that is exceeded, it send the message in chunks and then Graylog will re-assemble

rma-psmorris commented on cwms-servers/CWMS_HOME/config/system_config/cwms_system_logging.properties:
```diff
@@ -0,0 +1,61 @@
+#Logstash Log Handler Setup
+biz.paluch.logging.gelf.jul.GelfLogHandler.host = http:<host>
+biz.paluch.logging.gelf.jul.GelfLogHandler.port = <port>
+biz.paluch.logging.gelf.jul.GelfLogHandler.facility = "NWP CWMS v3.3"
```
What does this correspond to?

adamkorynta replied:
well, the documentation no longer says since it's deprecated. Removed

rma-psmorris reviewed (changes requested):
Questions and feedback

adamkorynta committed:
Update logging setup and process

The commit updates the logging setup in sendLogMessage, cwmsClasspath, logWatcher.py, and loggingSetup.py. These optimizations consist of removing the LOG_FACILITY variable, cleaning up comments, adding more meaningful constants, and improving logging paths. More changes have also been made specifically to standard output/error rerouting and improving sleep loops to use variable time intervals. Furthermore, the "logging_config.json" file has been renamed to "python_logging_config.json". Lastly, the logging level in "cwms_system_logging.properties" has been updated to "ALL".

adamkorynta committed:
Add comment descriptions to logging methods

rma-psmorris reviewed (approved):
All feedback resolved
